src/pages/index.js

```diff
@@ -1,103 +1,193 @@
 const profileForm = document.forms["modal-form"];
+const editImageForm = document.querySelector("#image-edit-form");
+const avatarImage = document.querySelector("#profile-avatar-image");
+const profileEditImageForm = document.querySelector("#image-edit-form");
+const pencil = document.querySelector("#profile-pencil");
 
 // Card Variables
 
 const addCardModal = document.querySelector("#add-card-modal");
 const addCardForm = addCardModal.querySelector("#add-card-form");
-const cardTitleInput = document.querySelector("#card-title-input");
-const cardUrlInput = document.querySelector("#modal-url-input");
 const cardsWrap = document.querySelector(".cards__list");
 
 // Preview Image Variables
 
 const previewModal = document.querySelector("#preview-modal");
 const modalImage = previewModal.querySelector(".modal__image");
-const previewCaption = previewModal.querySelector(".modal__caption");
 
 // Buttons
 
 const profileEditBtn = document.querySelector("#profile-edit-button");
 const addNewCardBtn = document.querySelector("#add-card-button");
-const closeBtns = document.querySelectorAll(".modal__close-button");
 
-// FUNCTIONS
+// API'S
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "250daeea-0b63-48f9-a4b3-e529116433c4",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    newCardSection.setItems(cards);
+    newCardSection.renderItems();
+  })
+  .catch((err) => {
+    alert(`${err} Failed to get cards.`);
+  });
+
+api
+  .getUserInfo()
+  .then((info) => {
+    profileUserInfo.setUserInfo({
+      name: info.name,
+      description: info.about,
+    });
+    profileUserInfo.setUserAvatar(info.avatar);
+  })
+  .catch((err) => {
+    alert(`${err} Failed to get user info.`);
+  });
+
+// API FUNCTIONS
 
 function handleProfileEditSubmit(data) {
-  profileUserInfo.setUserInfo(data);
+  profileEditForm.setLoading(true);
+  api
+    .updateUserInfo(data)
+    .then((result) => {
+      profileUserInfo.setUserInfo(data);
+      profileEditForm.close();
+    })
+    .catch((err) => {
+      alert(`${err} Failed to change user info.`);
+    })
+    .finally(() => {
+      profileEditForm.setLoading(false);
+    });
 }
 
 function handleCardSubmit({ title: name, subtitle: link }) {
-  newCardSection.addItem({ name, link });
-  addImageForm.close();
+  addImageForm.setLoading(true);
+  api
+    .addCard({ name, link })
+    .then((card) => {
+      newCardSection.addItem({ name, link });
+      addImageForm.close();
+    })
+    .catch((err) => {
+      alert(`${err} Failed to add card.`);
+    })
+    .finally(() => {
+      addImageForm.setLoading(false);
+    });
+}
+
+function handleAvatarSubmit(input) {
+  avatarEditForm.setLoading(true);
+  api
+    .changeAvatar(input.link)
+    .then((result) => {
+      profileUserInfo.setUserAvatar(result.avatar);
+      avatarEditForm.setLoading(false);
```

Comment:
Needs correction: Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally